dali/operators/image/crop/crop_mirror_normalize.cc

```diff
@@ -120,20 +119,30 @@ void CropMirrorNormalize<CPUBackend>::RunImpl(HostWorkspace &ws) {
         using Kernel = kernels::SliceFlipNormalizePermutePadCpu<OutputType, InputType, Dims>;
         using Args = kernels::SliceFlipNormalizePermutePadArgs<Dims>;
         auto &kernel_sample_args = any_cast<std::vector<Args>&>(kernel_sample_args_);
-        for (int sample_id = 0; sample_id < nsamples; sample_id++) {
-          thread_pool.AddWork(
-            [this, &input, &output, &kernel_sample_args, sample_id](int thread_id) {
-              auto in_view = view<const InputType, Dims>(input[sample_id]);
-              auto out_view = view<OutputType, Dims>(output[sample_id]);
-              auto &args = kernel_sample_args[sample_id];
-              kernels::KernelContext ctx;
-              kmgr_.Run<Kernel>(thread_id, sample_id, ctx, out_view, in_view, args);
-            }, out_shape.tensor_size(sample_id));
+        auto in_view = view<const InputType, Dims>(input);
+        auto out_view = view<OutputType, Dims>(output);
+
+        if (nsamples < thread_pool.NumThreads() * 4) {  // intra-sample parallelization
```

Comment:
Same for slice_base.

Reply:
The idea is to have enough blocks so that the variability between samples doesn't make the threads unbalanced, since the threads will be picking larger blocks first.